clouddriver-oracle/src/main/groovy/com/netflix/spinnaker/clouddriver/oracle/deploy/op/UpsertOracleLoadBalancerAtomicOperation.java

```diff
@@ -0,0 +1,332 @@
+ * You can obtain one at https://www.apache.org/licenses/LICENSE-2.0.html
+ */
+package com.netflix.spinnaker.clouddriver.oracle.deploy.op;
+
+import com.netflix.spinnaker.clouddriver.data.task.Task;
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository;
+import com.netflix.spinnaker.clouddriver.oracle.deploy.OracleWorkRequestPoller;
+import com.netflix.spinnaker.clouddriver.oracle.deploy.description.UpsertLoadBalancerDescription;
+import com.netflix.spinnaker.clouddriver.oracle.model.Details;
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation;
+import com.oracle.bmc.loadbalancer.model.Certificate;
+import com.oracle.bmc.loadbalancer.model.CertificateDetails;
+import com.oracle.bmc.loadbalancer.model.CreateCertificateDetails;
+import com.oracle.bmc.loadbalancer.model.BackendDetails;
+import com.oracle.bmc.loadbalancer.model.BackendSet;
+import com.oracle.bmc.loadbalancer.model.BackendSetDetails;
+import com.oracle.bmc.loadbalancer.model.CreateBackendSetDetails;
+import com.oracle.bmc.loadbalancer.model.CreateListenerDetails;
+import com.oracle.bmc.loadbalancer.model.CreateLoadBalancerDetails;
+import com.oracle.bmc.loadbalancer.model.ListenerDetails;
+import com.oracle.bmc.loadbalancer.model.LoadBalancer;
+import com.oracle.bmc.loadbalancer.model.UpdateBackendSetDetails;
+import com.oracle.bmc.loadbalancer.model.UpdateListenerDetails;
+import com.oracle.bmc.loadbalancer.requests.CreateCertificateRequest;
+import com.oracle.bmc.loadbalancer.requests.CreateListenerRequest;
+import com.oracle.bmc.loadbalancer.requests.CreateLoadBalancerRequest;
+import com.oracle.bmc.loadbalancer.requests.CreateBackendSetRequest;
+import com.oracle.bmc.loadbalancer.requests.DeleteCertificateRequest;
+import com.oracle.bmc.loadbalancer.requests.DeleteBackendSetRequest;
+import com.oracle.bmc.loadbalancer.requests.DeleteListenerRequest;
+import com.oracle.bmc.loadbalancer.requests.GetLoadBalancerRequest;
+import com.oracle.bmc.loadbalancer.requests.UpdateBackendSetRequest;
+import com.oracle.bmc.loadbalancer.requests.UpdateListenerRequest;
+import com.oracle.bmc.loadbalancer.responses.CreateBackendSetResponse;
+import com.oracle.bmc.loadbalancer.responses.CreateCertificateResponse;
+import com.oracle.bmc.loadbalancer.responses.CreateListenerResponse;
+import com.oracle.bmc.loadbalancer.responses.CreateLoadBalancerResponse;
+import com.oracle.bmc.loadbalancer.responses.DeleteBackendSetResponse;
+import com.oracle.bmc.loadbalancer.responses.DeleteCertificateResponse;
+import com.oracle.bmc.loadbalancer.responses.DeleteListenerResponse;
+import com.oracle.bmc.loadbalancer.responses.UpdateBackendSetResponse;
+import com.oracle.bmc.loadbalancer.responses.UpdateListenerResponse;
+import com.oracle.bmc.model.BmcException;
+import groovy.util.logging.Slf4j;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Slf4j
```

Comment:
I'd recommend against using the Groovy `@Slf4j` annotation in Java classes, just so there's less leakage of Groovy. Lombok has an annotation, or you can instantiate it manually.

Reply:
@Slf4j is not used in this class; removed. This class is using task.updateStatus for logging.